inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyClassElement.java

```diff
@@ -851,11 +851,22 @@ private List<GroovyPropertyElement> getPropertyNodes() {
             if (propertyNode.isPublic() && !propertyNode.isStatic()) {
                 final String propertyName = propertyNode.getName();
                 boolean readOnly = propertyNode.getField().isFinal();
+
+                ClassNode propDeclaringClass = propertyNode.getDeclaringClass();
+                GroovyClassElement declaringEl = this;
+                if (!propDeclaringClass.getName().equals(getName())) {
+                    declaringEl = new GroovyClassElement(
+                        visitorContext,
+                        propDeclaringClass,
+                        AstAnnotationUtils.getAnnotationMetadata(sourceUnit, compilationUnit, propDeclaringClass)
+                    );
+                }
+
                 final AnnotationMetadata annotationMetadata =
                         AstAnnotationUtils.getAnnotationMetadata(sourceUnit, compilationUnit, propertyNode.getField());
                 GroovyPropertyElement groovyPropertyElement = new GroovyPropertyElement(
```

Comment:
FYI most of the code in elements contains `declaredType` but that's actually owned type. I'm working on a PR that changes it.